### PR TITLE
Video handling rewrite that should've been PR'd a year ago

### LINKED ISSUE #2708: Data Saving for Videos
Slide only offers data saving for pictures. It would be great if we could get a similar setting for Gifs/Videos. 

Slide version: 5.8.5
Android version: 7.1.1

### LINKED ISSUE #2885: GIF/Video compression when streaming
Some gif/video files are greater than 50MB when streaming. Is it possible to compress them before streaming? Reddit videos are usually the culprits in my experience.

### LINKED ISSUE #2936: V.reddit videos don't play in peek mode
Slide version:6.0.1-3
Android version:8.1.0

V.reddit links don't play in peek mode. Long pressing v.reddit links opens  the peak viewer but nothing plays on it  They play fine if you click on them normally.

### LINKED ISSUE #2986: MP4 videos are tagged as GIF files making it so you can't mute them
Slide version: 6.0.1-3
Android version: 8.1.0 on LineageOS 15.1

I placed a request a few days ago (https://github.com/ccrama/Slide/issues/2979) but doing more troubleshooting it looks like it's an issue with MP4 videos incorrectly marked as GIF files.

Since it opens like a GIF file, there is no way to mute the video like any of the other video file formats.

### LINKED ISSUE #3083: Videos in Shadowbox mode autoplay with audio when not the focus
## Version Info
Slide v6.2.2

Android 9, 8

## Description
When viewing a subreddit that has multiple reddit videos in Shadowbox mode, videos play with audio as soon as they are loaded. This can result in audio from 2 videos; the active one on screen and the next one that has finished loading, playing at the same time.

## Steps to Replicate
1. Go to a subreddit with a lot of reddit video posts
1. Enter Shadowbox mode
1. Swipe left until you get a post with Reddit video
1. If it sounds like multiple videos are playing at the same time, pause the active one to verify

### LINKED ISSUE #3179: Shadow box mode
Slide version: 6.4
Android version: 9
Shadowbox mode doesn't support redgif.com links

### LINKED ISSUE #3196: No audio on some v.redd.it videos.
Slide version: 6.4
Android version: 11 Beta

Some v.redd.it videos do not play the audio. There is not even the mute icon if I tap the video to bring up the menu overlay. I noticed this change after upgrading to the Android 11 Beta.

The audio does play in a browser.

Example video: https://redd.it/hjb7gf

### LINKED ISSUE #3204: Some v.redd.it videos open without sound
Slide version: 6.4
Android version: 9

Some videos uploaded to v.redd.it are playing without sound. The affected videos are seemingly random. A large number of people are affected judging by this thread: https://www.reddit.com/r/slideforreddit/comments/hizr86/no_audio_for_some_vreddit_videos/

Two affected videos:
https://www.reddit.com/r/PublicFreakout/comments/hlqguz/drunk_florida_man_trumper_chasing_the_grim_reaper/

https://www.reddit.com/r/PublicFreakout/comments/hlsl2l/woman_gives_oral_sex_to_a_demon_in_church/

Two working videos:
https://www.reddit.com/r/PublicFreakout/comments/hlmusu/jacked_70_year_old_man_gets_roidroad_rage/

https://www.reddit.com/r/PublicFreakout/comments/hlpfyj/karen_goes_to_hospital_because_of_her_broken/

### LINKED ISSUE #3208: Gifs hosted on "that new gifhost" fail to load in slideshow view
When using slideshow view, gifs hosted on redgif are not loading. Just a black screen is shown

### PR DESCRIPTION
**Pull request by timawesomeness**
Fixes everything wrong with videos, like 

* **muting mp4s from imgur**! and anywhere else
* streamable
* silent vreddits
* stupidly muxing every single dash video ever
* literally caching every video twice
* redgifs in shadowbox
* whatever else i don't remember

Fixes #3208, fixes #3204, fixes #3196, fixes #3179, should fix #3083 iirc, fixes #2986, fixes #2936, fixes #2885, fixes #2708, fixes your broken arm, fixes your marriage, fixes everything wrong with the world

Also has starting points for zoomable video (#2652) and autoplay video (#3007) if I ever get around to either before the rewrite

### CODE REVIEW

File: app/src/main/java/me/ccrama/redditslide/Fragments/MediaFragment.java [hunk 1]
```diff
@@ -496,6 +479,13 @@ public void doLoadGif(final Submission s) {
                     .get("source")
                     .get("url")
                     .asText()).replace("&amp;", "&");
+        } else if ((t.shouldLoadPreview() && s.getDataNode().has("preview")
+                && s.getDataNode().get("preview").has("reddit_video_preview"))) {
+            toLoadURL = StringEscapeUtils.unescapeJson(s.getDataNode()
+                    .get("preview")
+                    .get("reddit_video_preview")
+                    .get("dash_url")
+                    .asText()).replace("&amp;", "&");
```
[the remark this answers is not in the document]
Comment (timawesomeness) [reply app/src/main/java/me/ccrama/redditslide/Fragments/MediaFragment.java]: actually that's not even needed there, i just copied and pasted

**Comment by ccrama** [comment]
This is some awesome work, I really appreciate the PR! 

Didn't know I was married but this PR fixed my marriage for sure...

**Comment by TacoTheDank** [comment]
@timawesomeness We really don't deserve you man, good job!

**Comment by LiquidZer0** [comment]
I see the update notes for 6.5, but cannot find an update on either the Google Play Store or F-Droid. Neither platform had 6.5 available to download.

**Comment by ccrama** [comment]
We are waiting on Google to re-approve Slide, once it’s back on the store you will be able to update.

**Comment by nebulaw2081d** [comment]
@ccrama Will F-Droid be updated to 6.5. I would hope re-approval is not needed there.

**Comment by Alexendoo** [comment]
@nebulaw2081d You can follow https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7114 for the progress on updating f-droid

**Pull request merged**